xarray/backends/zarr.py

```diff
@@ -784,7 +895,11 @@ def store(
             variables_to_set, check_encoding_set, writer, unlimited_dims=unlimited_dims
         )
         if self._consolidate_on_close:
-            zarr.consolidate_metadata(self.zarr_group.store)
+            kwargs = {}
+            if _zarr_v3():
+                # https://github.com/zarr-developers/zarr-python/pull/2113#issuecomment-2386718323
```

Comment:
I'll look more closely later, but for now I think this will be required, following a deliberate change in zarr v3 consolidated metadata.

With v2 metadata, *I think* that consolidated happened at the store-level,  and was all-or-nothing. If you have two Groups with Arrays, the consolidated metadata will be placed at the store root, and will contain everything:


```
# zarr v2

In [1]: import json, xarray as xr

In [2]: store = {}

In [3]: a = xr.tutorial.load_dataset("air_temperature")

In [4]: b = xr.tutorial.load_dataset("rasm")

In [5]: a.to_zarr(store=store, group="A")
/Users/tom/gh/zarr-developers/zarr-v2/.direnv/python-3.10/lib/python3.10/site-packages/xarray/core/dataset.py:2562: SerializationWarning: saving variable None with floating point data as an integer dtype without any _FillValue to use for NaNs
  return to_zarr(  # type: ignore[call-overload,misc]
Out[5]: <xarray.backends.zarr.ZarrStore at 0x11113edc0>

In [6]: b.to_zarr(store=store, group="B")
Out[6]: <xarray.backends.zarr.ZarrStore at 0x10cab2440>

In [7]: list(json.loads(store['.zmetadata'])['metadata'])
Out[7]:  # contains nodes from both A and B
['.zgroup',
 'A/.zattrs',
 'A/.zgroup',
 'A/air/.zarray',
 'A/air/.zattrs',
 'A/lat/.zarray',
 'A/lat/.zattrs',
 'A/lon/.zarray',
 'A/lon/.zattrs',
 'A/time/.zarray',
 'A/time/.zattrs',
 'B/.zattrs',
 'B/.zgroup',
 'B/Tair/.zarray',
 'B/Tair/.zattrs',
 'B/time/.zarray',
 'B/time/.zattrs',
 'B/xc/.zarray',
 'B/xc/.zattrs',
 'B/yc/.zarray',
 'B/yc/.zattrs']
```

With v3, consolidated metadata is scoped to a Group, so we *can* provide the group we want to consolidated (the zarr-python API does support "consolidate everything in the store at the root", but I don't think we want that because you'd need to open it at the root when reading, and I think it's kinda where for `ds.to_zarr(group="A")` to be reading / writing stuff outside of the `A` prefix).

Reply:
Agreed. https://github.com/zarr-developers/zarr-specs/pull/309 has some discussion on adding a `depth` field to the spec for consolidated metadata. That's currently implicitly `depth=None`, which is everything below a group. `depth=0` or `1` would be just the immediate children. That's not standardized or implemented anywhere yet, but the current implementation is forwards compatible and it shouldn't be a ton of effort.